### PR TITLE
Test refactoring to use let

### PR DESCRIPTION
**Pull request by dafeys**
dev

## ZeroWaste
- #811 

## Code reviewers

- [x] @loqimean 
- [x] @DmytroStoliaruk 

## Summary of issue
- Identify areas within the spec suite where variable instantiation can be optimized

## Summary of change
- Test refactoring to use let instead of variables
- Test refactoring to use route helper and shared context
- Add some tests

## CHECK LIST
- [x]  СI passed
- [x]  Сode coverage >=95%
- [x]  PR is reviewed manually again (to make sure you have 100% ready code)
- [x]  All reviewers agreed to merge the PR
- [x]  I've checked new feature as logged in and logged out user if needed
- [x]  PR meets all conventions

### CODE REVIEW

**Review by DmytroStoliaruk — APPROVED** [review]

File: spec/features/reset_password_spec.rb [hunk 1]
```diff
@@ -1,16 +1,16 @@
 # frozen_string_literal: true
 
 require "rails_helper"
-PASSWORD_RESET_PATH = "/users/password/new"
 
 xdescribe "Password Reset Page", js: true do
+  let(:password_reset_path) { "/users/password/new" }
```
Comment (loqimean) [inline spec/features/reset_password_spec.rb:6]: давай роут хелпер заюзаєш

File: spec/features/visit_admin_page_spec.rb [hunk 2]
```diff
@@ -3,23 +3,39 @@
 require "rails_helper"
 
 describe "visit Admin page", js: true do
+  let(:user) { create(:user) }
+
   before do
-    @user = create(:user)
-    sign_in @user
+    sign_in user
```
Comment (loqimean) [inline spec/features/visit_admin_page_spec.rb:9]: а на це хіба нема у нас шерид контексту?

File: spec/requests/diaper_calculators_spec.rb [hunk 3]
```diff
@@ -87,9 +90,11 @@
         }
       end
 
+      let(:expected_result) { { result: invalid_values } }
```
Comment (loqimean) [inline spec/requests/diaper_calculators_spec.rb:93]: так писати let не гуд, він завжди має бути визначений зверху

Comment (dafeys) [reply spec/requests/diaper_calculators_spec.rb:93]: Всі let в горі кожного context )

**Review by loqimean — APPROVED** [review]

**Pull request merged**